overlord/configstate/configcore/experimental.go

```diff
@@ -87,3 +88,27 @@ func doExportExperimentalFlags(_ sysconfig.Device, tr ConfGetter, opts *fsOnlyCo
 func ExportExperimentalFlags(tr ConfGetter) error {
 	return doExportExperimentalFlags(nil, tr, nil)
 }
+
+// CleanExperimentalFlags removes experimental flag configs that used to
+// exist but now are out of experimental.
+func CleanExperimentalFlags(tr *config.Transaction) error {
```

Comment:
This does mean that if we revert snapd then experimental features someone used to have enabled will no longer be enabled right?

Reply:
I think so

The aspect of reverting snapd is hard to get right.

Reply:
That's a good point, Yes. I didn't think of that, and in case of a snapd upgrade failure anything depending on a removed experimental flag could really break.